### PR TITLE
Ecommerce: recording multiple purchase items with the same sku but different values

### PR DESCRIPTION
**Pull request by Slowlearneruk**
Please issue pull request against the `3.x-dev` branch only.

Piwik 2 is in LTS mode. This means we do not accept any pull request for 2.x except critical security bugs and major data loss bugs. 

If you need to create a pull request for 2.x, then please also create the pull request against the `3.x-dev` so we can merge both.

Happy hacking!

### CODE REVIEW

**Comment by Slowlearneruk** [comment]
Hmm.. not very familiar with the submission procedure on Github.

In summary we are recording multiple purchase items with the same sku but different values on our site (these represent bets with different stakes on the same event). Consequently we need to permit an ecommerce request with multiple entries. As you can see, using an object with the sku as a key does not really work with this model but as it happens we can easily switch ecommerceItems to an array as the sku is also held in the item member.

File: js/piwik.js [hunk 1]
```diff
@@ -4193,42 +4193,21 @@ if (typeof window.Piwik !== 'object') {
...
 
-                if (ecommerceItems) {
-                    // Removing the SKU index in the array before JSON encoding
-                    for (sku in ecommerceItems) {
-                        if (Object.prototype.hasOwnProperty.call(ecommerceItems, sku)) {
-                            // Ensure name and category default to healthy value
-                            if (!isDefined(ecommerceItems[sku][1])) {
-                                ecommerceItems[sku][1] = "";
+                if (ecommerceItems.length) {
+                    ecommerceItems.forEach(function(ecommerceItem) {
```
Comment (tsteur) [inline js/piwik.js]: .forEach might not be supported in all browsers, the tracker is supposed to work even in older browsers. Can you maybe rewrite it to use a plain for loop?

**Comment by Slowlearneruk** [comment]
Done, what are your guidelines for browser support? Array.prototype.foreach seems to be covered by everything: http://kangax.github.io/compat-table/es5/#test-Array.prototype.forEach what am I missing?

Cheers, for feedback.

**Comment by tsteur** [comment]
We also support IE 8, IE7, IE6 ,...

**Comment by mattab** [comment]
I also gave this a try few months ago in https://github.com/piwik/piwik/pull/10297/files but then realised it is not that simple because At the moment in Piwik the Tracking API backend does expect at most one product with a given SKU, as there is a `PRIMARY KEY(idvisit, idorder, idaction_sku),` on `log_ecommerce_item` table...

**Comment by mattab** [comment]
If someone ever wants to work on this please re-create the PR and we can help with reviews etc. For us it's not a priority at this time

**Pull request closed without merging**